pages/k8s/cni-multus.md

```diff
@@ -0,0 +1,225 @@
+---
+wrapper_template: "kubernetes/docs/base_docs.html"
+markdown_includes:
+  nav: "kubernetes/docs/shared/_side-navigation.md"
+context:
+  title: "CNI with Multus"
+  description: How to deploy and use Multus in Charmed Kubernetes
+keywords: CNI, networking, multus
+tags: [operating]
+sidebar: k8smain-sidebar
+permalink: cni-multus.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+[Multus][multus] is a CNI plugin for Kubernetes that enables attaching multiple
+network interfaces to pods. Multus support for **Charmed Kubernetes** is
+provided as a Kubernetes charm.
+
+## Requirements
+
+### Juju 2.8 features
+
+The Multus charm requires functionality from Juju 2.8+, which is currently
+under active development. In order to run Multus, you will need a Juju
+controller running Juju 2.8 from edge.
+
+Install Juju 2.8 from edge:
+
+```
+sudo snap install juju --channel edge --classic
+```
+
+Or if Juju is already installed, refresh it:
+
+```
+sudo snap refresh juju --channel edge
+```
+
+Once you have the Juju 2.8 edge client, any new controllers you bootstrap will
+include the 2.8 features required by Multus.
+
+### Persistent volume support
+
+In order to deploy Multus, the first requirement is that you must have
+persistent volume support enabled in your **Charmed Kubernetes** cluster.
+
+If your cluster includes any of the cloud integrator charms, then you should
+have persistent volume support already, and can skip this section.
+
+Otherwise, you can read the [Storage][storage] documentation page to learn how
+to enable persistent volume support by adding Ceph to your cluster.
+
+### Creating a Kubernetes model in Juju
+
+Multus support for **Charmed Kubernetes** is provided as a Kubernetes charm,
+which means that you will need to create a Kubernetes model in Juju.
+
+First, make sure your local kubeconfig is pointing to the correct Kubernetes
+cluster:
+
+```
+juju scp kubernetes-master/0:config ~/.kube/config
+```
+
+Next, add your Kubernetes as a cloud to your Juju controller:
+
+```
+juju add-k8s my-k8s-cloud --controller $(juju switch | cut -d: -f1)
+```
+
+And create a new Kubernetes model:
+
+```
+juju add-model my-k8s-model my-k8s-cloud
+```
+
+## Deploying Multus
+
+Once all of the requirements have been met, you can deploy Multus into a
+Kubernetes model by running:
+
+```
+juju deploy cs:~containers/multus
```

Comment:
This command won't work until we get Multus pushed to cs:~containers. I'll work on that ASAP.

Reply:
This command should work now, though I still need to test it. I've pushed the charm to cs:~containers/multus and promoted through edge, beta, candidate, and stable. Granted read access to "everyone".